File: frontend/types/page.ts
export type PageType = "home" | "about" | "projects" | "blog" | "contact";
export type PageStatus = "draft" | "scheduled" | "published" | "unpublished";

export interface SeoMeta {
  id: string;
  seo_title: string | null;
  meta_description: string | null;
  focus_keyword: string | null;
  keywords: string[] | null;
  canonical_url: string | null;
  og_title: string | null;
  og_description: string | null;
  og_image_media_id: string | null;
  twitter_card_type: string | null;
  twitter_title: string | null;
  twitter_description: string | null;
  twitter_image_media_id: string | null;
  robots: string | null;
  schema_jsonld: Record<string, unknown> | null;
}

export interface BlockDefinition {
  id: string;
  key: string;
  label: string;
  is_active: boolean;
}

export interface PageBlock {
  id: string;
  block_definition_id: string;
  position: number;
  config: Record<string, unknown>;
}

export interface Page {
  id: string;
  page_type: PageType;
  slug: string;
  title: string;
  status: PageStatus;
  published_at: string | null;
  scheduled_at: string | null;
  seo: SeoMeta | null;
  blocks: PageBlock[];
  updated_at: string;
}

export interface PageListItem {
  id: string;
  page_type: PageType;
  slug: string;
  title: string;
  status: PageStatus;
  updated_at: string;
}

export interface EntityVersion {
  id: string;
  version_number: number;
  change_note: string | null;
  created_by: string;
  created_at: string;
}

File: backend/app/models/page.py
import enum
import uuid
from datetime import datetime

from sqlalchemy import DateTime, Enum, ForeignKey, String

from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.shared.database.base import Base, TimestampMixin, UUIDPrimaryKeyMixin


class PageType(str, enum.Enum):
    HOME = "HOME"
    ABOUT = "ABOUT"
    PROJECTS = "PROJECTS"
    BLOG = "BLOG"
    CONTACT = "CONTACT"
    CUSTOM = "CUSTOM"


class PageStatus(str, enum.Enum):
    DRAFT = "DRAFT"
    SCHEDULED = "SCHEDULED"
    PUBLISHED = "PUBLISHED"
    UNPUBLISHED = "UNPUBLISHED"


class Page(UUIDPrimaryKeyMixin, TimestampMixin, Base):
    """Fixed to exactly 5 rows — `page_type` is a unique enum, not a free slug,
    so the "5 fixed pages" scope decision is enforced at the DB level, not just
    in application code (ERD.md).
    """

    __tablename__ = "pages"

    page_type: Mapped[PageType] = mapped_column(
        "pageType", Enum(PageType, name="PageType", create_type=False), unique=True, nullable=False
    )
    slug: Mapped[str] = mapped_column(String(100), unique=True, nullable=False)
    title: Mapped[str] = mapped_column(String(255), nullable=False)
    status: Mapped[PageStatus] = mapped_column(
        Enum(PageStatus, name="PageStatus", create_type=False), default=PageStatus.PUBLISHED, nullable=False
    )
    published_at: Mapped[datetime | None] = mapped_column("publishedAt", DateTime(timezone=True), default=None)
    scheduled_at: Mapped[datetime | None] = mapped_column("scheduledAt", DateTime(timezone=True), default=None)

    featured_image_media_id: Mapped[str | None] = mapped_column("featuredImageId", String(255), default=None)

    seo_id: Mapped[str | None] = mapped_column("seoId", String(255), ForeignKey("seo_meta.id"), default=None)
    created_by: Mapped[str | None] = mapped_column("createdById", String(255), ForeignKey("users.id"), default=None)
    updated_by: Mapped[str | None] = mapped_column("updatedById", String(255), ForeignKey("users.id"), default=None)

    seo: Mapped["SeoMeta | None"] = relationship()  # noqa: F821
    blocks: Mapped[list["PageBlock"]] = relationship(  # noqa: F821
        back_populates="page", order_by="PageBlock.position", cascade="all, delete-orphan"
    )
